addons/account_debit_note/wizard/account_debit_note.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+from odoo import models, fields, api
+from odoo.tools.translate import _
+
+
+class AccountDebitNote(models.TransientModel):
+    """
+    Add Debit Note wizard: when you want to correct an invoice with a tiny amount of something
+    """
+    _name = 'account.debit.note'
+    _description = 'Add Debit Note wizard'
+
+    move_id = fields.Many2one('account.move', string='Journal Entry',
+        domain=[('state', '=', 'posted'), ('type', 'not in', ('out_refund', 'in_refund'))])
+    date = fields.Date(string='Adjustment date', default=fields.Date.context_today, required=True)
+    reason = fields.Char(string='Reason')
+    journal_id = fields.Many2one('account.journal', string='Use Specific Journal', help='If empty, uses the journal of the journal entry to be reversed.')
+
+    # computed fields
+    residual = fields.Monetary(compute="_compute_from_moves")
+    currency_id = fields.Many2one('res.currency', compute="_compute_from_moves")
+    move_type = fields.Char(compute="_compute_from_moves")
+
+    @api.model
+    def default_get(self, fields):
+        res = super(AccountDebitNote, self).default_get(fields)
+        move_ids = self.env['account.move'].browse(self.env.context['active_ids']) if self.env.context.get('active_model') == 'account.move' else self.env['account.move']
+        res['residual'] = len(move_ids) == 1 and move_ids.amount_residual or 0
+        res['currency_id'] = len(move_ids.currency_id) == 1 and move_ids.currency_id.id or False
+        res['move_type'] = len(move_ids) == 1 and move_ids.type or False
+        return res
+
+    @api.depends('move_id')
+    def _compute_from_moves(self):
+        move_ids = self.env['account.move'].browse(self.env.context['active_ids']) if self.env.context.get('active_model') == 'account.move' else self.move_id
+        for record in self:
+            record.residual = len(move_ids) == 1 and move_ids.amount_residual or 0
+            record.currency_id = len(move_ids.currency_id) == 1 and move_ids.currency_id or False
+            record.move_type = len(move_ids) == 1 and move_ids.type or False
+
+    def _prepare_default_values(self, move):
+        return {
+                'ref': _('Adjustment of: %s, %s') % (move.name, self.reason) if self.reason else _('Adjustment of: %s') % (move.name),
+                'date': self.date or move.date,
+                'invoice_date': move.is_invoice(include_receipts=True) and (self.date or move.date) or False,
+                'journal_id': self.journal_id and self.journal_id.id or move.journal_id.id,
+                'invoice_payment_term_id': None,
+                'debit_origin_id': move.id,
+            }
+
+    def create_debit(self):
+        move = self.env['account.move'].browse(self.env.context['active_ids']) if self.env.context.get('active_model') == 'account.move' else self.move_id
+        move.ensure_one()
```

Comment:
This is breaking calling the wizard with multiple invoices

Reply:
In which case a debit note is related to more than 1 invoice? AFAICR debit note by definition is the extension of 1 invoice only!

Reply:
In argentina you can. One debit/credit not can be from many invoices.
Web service support it, afip online invoicing support it and you can check it [here](https://www.afip.gob.ar/noticias/20190801-notas-de-credito-y-o-debito.asp)

"...siempre que se encuentren relacionadas a una o más facturas o documentos equivalentes emitidos previamente..."

Reply:
Ok, that make sense.

Reply:
@jco-odoo 

Hello, @nhomar normally a debit note is intended for one invoice, or one debit note also. Nevertheless, there could be a case where you use a debit note to surcharge interests over an amount due integrated by more than one invoice.

Particularly I agree with almost all @jjscarafia points. In l10n_cl case, (in l10n_cl_edi) we will need to refer the debit note originating documents in the current 'account.invoice.reference' model.

Perhaps we never will be generating a debit notes for documents that are located in different journals. Thus in the wizard, the default journal should be the wizard from the originating invoices or credit notes.

There will be also a need for a "reason" of the debit note. This is used as a description in the references.

As you also know there are 3 types of debit notes for l10n_cl:
- full reverse of a credit note: in this case all the items in the debit note should be the same as the credit note, with the very same amount and taxes if they existed originally.
- modify the amount: (surcharge over an invoice e.g. interests or change in the price)
- modify the text of a credit note: (0 amount, but a description should contain the fe de erratas text saying what fields should say something different in the invoice).

Regarding the conciliation: the only case of direct conciliation should be for a credit note.